### ISSUE TITLE
Remove trailing slash from void HTML elements

### ISSUE THREAD

**Issue by ahukkanen** [rails/rails#51643]
### Steps to reproduce

1. Follow https://guides.rubyonrails.org/getting_started.html up until the generation of the first view of the application
2. Look at the page source
3. See void elements using trailing slashes, e.g. `<meta name="csrf-param" content="authenticity_token" />`
4. Copy the source of this page (using view-source)
5. Paste the source to https://validator.w3.org/nu/#textarea
6. See a lot of "info" level notices for trailing slashes on the void elements

Recently the W3C validator is giving a lot of the following types of "info" level notices for any markup generated by a Rails application:
![W3C notices on trailing slashes for void elements](https://github.com/rails/rails/assets/864340/2cb080a7-6330-48d9-b4da-953e4e45843f)

Or in text format:

> Trailing slash on void elements [has no effect](https://github.com/validator/validator/wiki/Markup-%C2%BB-Void-elements#trailing-slashes-in-void-element-start-tags-do-not-mark-the-start-tags-as-self-closing) and [interacts badly with unquoted attribute values](https://github.com/validator/validator/wiki/Markup-%C2%BB-Void-elements#trailing-slashes-directly-preceded-by-unquoted-attribute-values).

### Expected behavior
I would expect to be able to generate a page source that would give me a clean W3C HTML validation report without any notices.

While both formats are valid, using the trailing slash is discouraged and causes a lot of noise in the validation report.

### Actual behavior
I get a lot of the described notices in the W3C HTML validation report.

If there is a reason (unknown to me) to preserve the current behavior, I would make it configurable in `config.action_view` to revert to the old behavior.

As far as I know, this originates from these two lines at `ActionView::TagHelper`:

https://github.com/rails/rails/blob/734719d8af033062cb7088a1353be6276659e3c1/actionview/lib/action_view/helpers/tag_helper.rb#L248

https://github.com/rails/rails/blob/734719d8af033062cb7088a1353be6276659e3c1/actionview/lib/action_view/helpers/tag_helper.rb#L497

And in addition, there are also few matches at `rails-ujs.js` and `rails-ujs.esm.js`:

https://github.com/rails/rails/blob/734719d8af033062cb7088a1353be6276659e3c1/actionview/app/assets/javascripts/rails-ujs.js#L406-L410

https://github.com/rails/rails/blob/734719d8af033062cb7088a1353be6276659e3c1/actionview/app/assets/javascripts/rails-ujs.esm.js#L454-L458

### System configuration
**Rails version**: pretty much any version, e.g. 7.1.3.2

**Ruby version**: 3.3.0

**Comment by Earlopain** [comment]
Yes, this is something I noticed a few days ago as well. There are also a few other places where this is the case like error pages. You can utilize a regex search to find most of these cases.

rails-ujs however is deprecated, see https://github.com/rails/rails/pull/50535 and https://github.com/rails/rails/pull/50555, so I doubt this will be changed for that.

**Comment by ahukkanen** [comment]
I made a PR (#51646) to solve the issue. Will look into the test results to see any unexpected implications, at least the `actionview` tests seemed to pass after the changes.

Most other parts were in tests and I also found one actual place where the trailing slashes are needed (elements inside SVGs) but these seemed to be covered by the tests.
